src/node.js

```diff
@@ -335,23 +335,27 @@
         scheduleMicrotasks();
     }
 
+    function _tryCallback(callback) {
+      var threw = true;
+      try {
+        callback();
+        threw = false;
+      } finally {
+        if (threw)
+          tickDone();
```

Comment:
catching and rethrowing is not the same as not catching in the first place, when it reaches fatal error the usual reporter that highlights the line that threw the error would become useless.